File: docs/WidgetMill/packages/spec/src/nodes/positioned.ts
import { z } from "zod";
import { NodeRef } from "../node-ref";

/** Positioned — sub-tipo de filho do Stack (spec §2.4). */
export const PositionedNode = z.object({
  type: z.literal("positioned"),
  props: z
    .object({
      left: z.number().optional(),
      top: z.number().optional(),
      right: z.number().optional(),
      bottom: z.number().optional(),
      width: z.number().optional(),
      height: z.number().optional(),
    })
    .default({}),
  child: NodeRef.optional(),
});

File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/positioned.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// Positioned — spec §2.4 (filho de Stack).
Widget buildPositioned(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return Positioned(
    left: parseDouble(p['left']),
    top: parseDouble(p['top']),
    right: parseDouble(p['right']),
    bottom: parseDouble(p['bottom']),
    width: parseDouble(p['width']),
    height: parseDouble(p['height']),
    child: r.maybeRender(context, node.child) ?? const SizedBox.shrink(),
  );
}
